sympy/solvers/solveset.py

```diff
@@ -1754,6 +1754,86 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _is_lambert(f, symbol):
+    r"""
+    The below 3 conditions are currently considered as Lambert type:
```

Comment:
What does "considered as Lambert type" mean?

Does it mean that this equation can definitely be solved using the Lambert W function? If so then that should be stated explicitly. Otherwise the significance of "Lambert type" should be explained.

Reply:
> Otherwise the significance of "Lambert type" should be explained.


Lambert type : The equation whose solution(s) will be in terms of LambertW function.

Is it ok to add these lines at `L-1759`.

Reply:
> Lambert type : The equation whose solution(s) will be in terms of the LambertW function.


Please let me know if there are any more modifications to this definition so that I may proceed to a final commit : )

Reply:
I don't think that this function returning true guarantees that the solution could or should be represented using Lambert W.

Do you mean that the solution "*might* be expressible using Lambert W"?

What does it mean if this function returns False?

Does it really mean "we will try to use the Lambert solver if this is true and it may or may not work"?

Reply:
> I don't think that this function returning true guarantees that the solution could or should be represented using Lambert W.

Yes, if it returns `true` it means that the solution may return a Lambert solution (may be other solutions are possible also if present) .


> What does it mean if this function returns False?

If it returns `false`  there is no possibility of getting any LambertW (function) solution.

> Does it really mean "we will try to use the Lambert solver if this is true and it may or may not work"?

Yes, sometimes the solutions will not be returned (instead `conditionset` in such cases mostly)... 

> Do you mean that the solution "might be expressible using Lambert W"?


Here `_is_lambert` means it will solve using `_solve_lambert` And mostly it try to solve and return solution(s) interms of Lambert... 


I hope I've answered your question and may be somewhere I may be wrong(or something I've missed) ... So, IMHO can you rather suggest something that exactly fits situation here...

Reply:
Okay, I would describe this as
```
Quick check for cases that the Lambert solver might be able to handle.

If this returns False then the Lambert solver will not be called.
```

Reply:
Nice :)